src/main/java/org/opensearch/timeseries/transport/BaseSuggestConfigParamTransportAction.java

```diff
@@ -82,8 +83,11 @@ public BaseSuggestConfigParamTransportAction(
         this.clock = Clock.systemUTC();
         this.context = context;
         this.searchFeatureDao = searchFeatureDao;
-        List<SuggestName> allSuggestParams = Arrays.asList(SuggestName.values());
-        this.allSuggestParamStrs = Name.getListStrs(allSuggestParams);
+        if (context.isAD()) {
+            this.allSuggestParamStrs = Name.getListStrs(Arrays.asList(ADSuggestName.values()));
+        } else {
+            this.allSuggestParamStrs = Name.getListStrs(Arrays.asList(ForecastSuggestName.values()));
+        }
```

Comment:
can we figure this out from the subclass? I want to avoid timeseries package class aware of subpackage classes if possible. So in base class, you leave allSuggestParamStrs as a parameter. 

BaseSuggestConfigParamTransportAction (...
List<SuggestName> allSuggestParams,
...)

in AD subclass, you pass in

super(...
Name.getListStrs(Arrays.asList(ADSuggestName.values()))
...)

Reply:
made this change